pkg/runner/operations/create.go

```diff
@@ -26,25 +27,33 @@ func operationCreate(ctx context.Context, obj ctrlclient.Object, c client.Client
 			logger.Log(operation, color.BoldRed, fmt.Sprintf("ERROR\n%s", _err))
 		}
 	}()
+
 	return wait.PollUntilContextCancel(ctx, interval, false, func(ctx context.Context) (bool, error) {
 		var actual unstructured.Unstructured
 		actual.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
 		err := c.Get(ctx, client.ObjectKey(obj), &actual)
 		if err == nil {
 			return false, errors.New("the resource already exists in the cluster")
 		} else if kerrors.IsNotFound(err) {
-			if err := c.Create(ctx, obj); err != nil {
+			createOptions := []ctrlclient.CreateOption{}
+			if dryRun {
+				createOptions = append(createOptions, ctrlclient.DryRunAll)
+			}
+			if err := c.Create(ctx, obj, createOptions...); err != nil {
 				if shouldFail {
 					return true, nil
 				}
 				return false, err
 			} else {
-				if cleaner != nil {
+				if cleaner != nil && !dryRun {
 					cleaner(obj, c)
 				}
 				if shouldFail {
 					return false, errors.New("an error was expected but didn't happen")
 				}
+				if dryRun {
```

Comment:
Do we need a dedicated log ?

Reply:
I think it would be better not a bad idea to have this